### DISCUSSION TITLE
Installation issues with CoolMUC2

### DISCUSSION THREAD

**Discussion by vikaskurapati** [SeisSol/SeisSol#828] [Q&A] [answered]
Hello,

I am trying to install SeisSol on CoolMUC2 cluster and I am unable to figure an error with building. After using the cmake command from the documentation, 

`CC=mpicc CXX=mpiCC FC=mpif90 cmake -DCOMMTHREAD=ON -DNUMA_AWARE_PINNING=ON -DASAGI=ON -DCMAKE_BUILD_TYPE=Release -DHOST_ARCH=skx -DPRECISION=double -DORDER=4 -DGEMM_TOOLS_LIST=LIBXSMM,PSpaMM ..`

I get the following error:

`CMake Error at /dss/dsshome1/lrz/sys/spack/release/22.2.1/opt/x86_64/cmake/3.21.4-gcc-6d6yeai/share/cmake-3.21/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find METIS (missing: METIS_LIBRARY METIS_INCLUDE_DIR
  METIS_TEST_RUNS)
Call Stack (most recent call first):
  /dss/dsshome1/lrz/sys/spack/release/22.2.1/opt/x86_64/cmake/3.21.4-gcc-6d6yeai/share/cmake-3.21/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  cmake/FindMETIS.cmake:80 (find_package_handle_standard_args)
  CMakeLists.txt:245 (find_package)
`

It seems like it is unable to find the metis_library and metis_include_dir but as instructed in the documentation, I checked and ensured that $HOME/include contains metis.h and $HOME/lib contains libmetis.a. 

I am not sure if this is the issue or if there is some other error in the installation. Any help is appreciated.

**Comment by Thomas-Ulrich** [comment]
I would suggest using the spack workflow.

**Reply by krenzland** [reply]
Is this currently working? There is still this open PR: https://github.com/SeisSol/seissol-spack-aid/pull/10

https://github.com/SeisSol/seissol-spack-aid/tree/supermuc_NG is maybe required on CoolMUC2 as well?

Regarding your issue: 
When installing the dependencies manually, you may need to set the following variables:
```
export CMAKE_PREFIX_PATH=~:$CMAKE_PREFIX_PATH
export PKG_CONFIG_PATH=~/lib/pkgconfig/:$PKG_CONFIG_PATH
```
as written here: https://seissol.readthedocs.io/en/latest/installing-dependencies.html?highlight=spack#initial-adjustments-to-bashrc

**Reply by vikaskurapati** [reply]
I have done the changes mentioned in bashrc from the link earlier and it didn't work. I put these two extra lines and now it successfully compiles. Even though it compiles, the executable fails to run on the interactive bash run(used salloc --ntasks=28). With the `./SeisSol_proxy_Release_dskx_4_elastic 100 100 all`, it gives the following error now:
`Illegal instruction (core dumped)`

With the compilation, `SeisSol_Release_dskx_4_elastic` and `SeisSol_proxy_Release_dskx_4_elastic` are created and I think these are the right ones for the cluster?

I will probably try the procedure Sebastian has posted underneath.

**Reply by davschneller** [reply]
The `skx` host arch adds some AVX-512 instructions (as supported on some Skylake CPUs) which are not supported by the CoolMUC2 CPUs (Haswell arch). The host arch `hsw` should work, e.g. you can try the option `-DHOST_ARCH=hsw`.

**Comment by sebwolf-de** [comment, marked as answer]
You can try this procedure: https://gist.github.com/sebwolf-de/0ebe30c6949d255797e353a9db756068

**Reply by vikaskurapati** [reply]
This works. Thank you very much!
